onnxruntime/core/providers/shared_library/provider_interfaces.h

```diff
@@ -0,0 +1,480 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// Public wrappers around internal ort interfaces (currently)
+// In the future the internal implementations could derive from these to remove the need for the wrapper implementations
+
+#include "core/framework/func_api.h"
+
+namespace ONNX_NAMESPACE {
+enum AttributeProto_AttributeType : int;
+enum OperatorStatus : int;
+
+// String pointer as unique TypeProto identifier.
+using DataType = const std::string*;
+
+struct Provider_TensorProto {
+  virtual ~Provider_TensorProto() = default;
+
+  virtual void CopyFrom(const Provider_TensorProto& v) = 0;
+
+  void operator=(const Provider_TensorProto& v) { CopyFrom(v); }
+};
+
+struct Provider_AttributeProto {
+  static std::unique_ptr<Provider_AttributeProto> Create();
+
+  virtual ~Provider_AttributeProto() = default;
+  virtual std::unique_ptr<Provider_AttributeProto> Clone() const = 0;
+
+  virtual ::onnx::AttributeProto_AttributeType type() const = 0;
+  virtual int ints_size() const = 0;
```

Comment:
should this use ONNX_NAMESPACE?

Reply:
actually given it's in namespace ONNX_NAMESPACE does it need the '::onnx::' prefix at all? 

---
In reply to: [419798034](https://github.com/microsoft/onnxruntime/pull/3314#discussion_r419798034) [](ancestors = 419798034)

Reply:
Looks like it didn't. I forget if it just had that when I first copied it. Given I override it with a new type it works without it.